backend/src/main/scala/com/softwaremill/adopttapir/infrastructure/Json.scala

```diff
@@ -1,12 +1,23 @@
 package com.softwaremill.adopttapir.infrastructure
 
 import com.softwaremill.tagging.@@
-import io.circe.generic.AutoDerivation
 import io.circe.{Decoder, Encoder, Printer}
+import io.circe.generic.extras.{AutoDerivation, Configuration => CirceConfiguration}
+import sttp.tapir.generic.{Configuration => TapirConfiguration}
 
 /** Import the members of this object when doing JSON serialisation or deserialisation.
   */
 object Json extends AutoDerivation {
+
+  implicit lazy val circeConfiguration: CirceConfiguration =
+    CirceConfiguration.default.withDiscriminator("type")
+      .copy(transformConstructorNames = (s => s.toLowerCase)
+    )
+
+  implicit val tapirConfiguration: TapirConfiguration =
+    TapirConfiguration.default.withDiscriminator("type")
+      .copy(toDiscriminatorValue = _.fullName.split('.').last.stripSuffix("$").toLowerCase)
```

Comment:
this looks like the default `Configuration:51`?

Reply:
yes, the only difference is to lower case conversion

Reply:
then it's best to reuse existing by chaining functions (`Configuration.default.toEncodedName.andThen(...)`)